python/tank/util/git.py

```diff
@@ -40,10 +42,25 @@ def execute_git_command(cmd):
         finally:
             os.chdir(cwd)
 
+    .. warning:: This method will not handle https repositories that use
+                 credentials. If no external authentication method
+                 has been defined, the method will fail with an error message,
+                 typically saying ``Cannot read Username for https://sitename``.
+                 In order to use such repositories with toolkit, we strongly
+                 recommend using ssh urls. If this is not an option, you need to
+                 specify an external authentication mechanism, for example::
+
+                    git config --system core.askpass git-gui--askpass
+
+                 For more information, see https://git-scm.com/docs/gitcredentials.
+
     :raises: Will raise a TankGitError on failure
     :param cmd: git command to execute (e.g. 'clone foo.git')
+    :returns: the output generated by the command
     """
+    log.debug("Attempting to execute git command '%s'" % cmd)
     # first probe to check that git exists in our PATH
+    log.debug("Checking that git exists and can be executed...")
     try:
         _ = subprocess_check_output("git --version", shell=True)
```

Comment:
Since you don't seem to be using shell features, please do not set shell=True to avoid shell injection.